compiler/semantic/sql.go

```diff
@@ -18,11 +18,20 @@ import (
 // and may or may not have its sources embedded.
 func (a *analyzer) semSelect(sel *ast.Select, seq dag.Seq) dag.Seq {
 	if sel.From != nil {
-		if len(seq) > 0 {
-			a.error(sel, errors.New("SELECT cannot have both an embedded FROM claue and input from parents"))
-			return append(seq, badOp())
+		off := len(seq)
+		hasParent := off > 0
+		seq = a.semFrom(sel.From, seq)
+		// If we have parents with both a from and select, report an error
+		// but only if it's not a RobotScan where the parent needs to feed
+		// the from.
+		if off < len(seq) {
```

Comment:
`off < len(seq)` will never be true here because `off := len(seq)`.

Reply:
semFrom is called between off getting set and getting checked.  The idea here was to protect the seq[off] check if semFrom didn't lengthen the chain.  I will rework it so it's more clear to return of the chain isn't lengthened.

Reply:
Completely missed that. 🤦